test/dutchExchange-claimBuyerFunds.js

```diff
@@ -0,0 +1,347 @@
+    const [num, den] = (await dx.getPriceForJS.call(gno.address, eth.address, auctionIndex)).map(i => i.toNumber())
+    let sellVolume = (await dx.sellVolumesCurrent.call(gno.address, eth.address))
+    let buyVolume = await dx.buyVolumes.call(gno.address, eth.address)
+    let oustandingVolume = (sellVolume.mul(bn(num)).div(den)).sub(buyVolume)
+    logger('oustandingVolume', oustandingVolume.toNumber())
+    logger('buyVolume', buyVolume)
+    assert.equal((bn(valMinusFee(totalSellAmount2ndAuction)).mul(buyVolume).div(buyVolume.add(oustandingVolume))).toNumber(), claimedAmount)
+
+    // actual testing at time with previous 2/3price
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 2 / 3)
+    const [claimedAmount2] = (await dx.claimBuyerFunds.call(gno.address, eth.address, buyer2, auctionIndex)).map(i => i.toNumber())
+    const [num2, den2] = await dx.getPriceForJS.call(gno.address, eth.address, auctionIndex)
+    sellVolume = (await dx.sellVolumesCurrent.call(gno.address, eth.address))
+    buyVolume = (await dx.buyVolumes.call(gno.address, eth.address))
+    oustandingVolume = (sellVolume.mul(num2).div(den2)).sub(buyVolume)
+    logger('oustandingVolume', oustandingVolume)
+    logger('buyVolume', buyVolume)
+    assert.equal((bn(valMinusFee(totalSellAmount2ndAuction)).mul(buyVolume).div(buyVolume.add(oustandingVolume))).toNumber(), claimedAmount2)
+  })
+})
+
+contract('DutchExchange - claimBuyerFunds', (accounts) => {
+  const [, seller1, buyer1] = accounts
+  const totalSellAmount2ndAuction = 10e18
+
+  before(async () => {
+    // get contracts
+    await setupContracts()
+
+    // set up accounts and tokens[contracts]
+    await setupTest(accounts, contracts, startBal)
+
+    // add tokenPair ETH GNO
+    await dx.addTokenPair(
+      eth.address,
+      gno.address,
+      10e18,
+      0,
+      2,
+      1,
+      { from: seller1 },
+    )
+
+    eventWatcher(dx, 'Log', {})
+  })
+
+  after(eventWatcher.stopWatching)
+
+  it('5. check right amount of coins is returned by claimBuyerFunds if auction is  not closed, but closed theoretical ', async () => {
+    // prepare test by starting and clearning new auction
+  
+    const auctionIndex = await getAuctionIndex()
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 1)
+    await postBuyOrder(eth, gno, auctionIndex, 10e18, buyer1)
+  
+
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 0.4)
+
+    // checking that closingPriceToken.num == 0
+    const [closingPriceNumToken] = (await dx.closingPrices(eth.address, gno.address, auctionIndex)).map(i => i.toNumber())
+    assert.equal(closingPriceNumToken, 0)
+    
+    // actual testing
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(eth.address, gno.address, buyer1, auctionIndex)).map(i => i.toNumber())
+    assert.equal(bn(valMinusFee(totalSellAmount2ndAuction)), claimedAmount)
```

Comment:
Shouldn't wrap `valMinusFee(totalSellAmount2ndAuction)` in `bn()`. Assert doesn't fail here thanks to loose equality conversion, but it won't work in all cases.